datafusion/functions/src/regex/regexpcount.rs

```diff
@@ -395,7 +394,7 @@ where
                     let pattern =
                         compile_and_cache_regex(regex, flags, &mut regex_cache)?;
 
-                    count_matches(value, &pattern, start)
+                    count_matches(value, pattern, start)
```

Comment:
All `Int64Array::from_iter_values` can be replaced by `.collect::<Result<Int64Array, ArrowError>>()?
This avoids one temporary Vec.

Reply:
Done.